### PR TITLE
fix: upgrade to TX-EDC 0.5.3 and resolve trivy/veracode issues.

### PR DESCRIPTION
**Pull request by drcgjung**
## WHAT

This PR upgrades the KA agent planes to depend on TX-EDC 0.5.3 (upcoming CX 23.12) and EDC 0.2.1
It furthermore resolves all open veracode issues that have accumulated until today.
Instead of fixing the base image, we allow trivy findings of CRITITCAL and HIGH vulnerability to proceed the action.
Finally it updates the version of this product and its modules and charts and integrates it with the other repos into a single Tractus-X product.

## WHY

trivy and veracode issues have accumulated.
In order to resolve them, we first upgrade the main (nested) dependencies and then look for appropriate patches to the still vulnerable artifacts (jars).
Finally we upgrade the version in order to prepare the next release (23.12) such that all Agent-Related repos appear under a leading umbrella.

## FURTHER NOTES

(Jar) Dependencies have been update and checked.
The base image vulnerabilities are not touched because of legal puposes.

### CODE REVIEW

**Comment by drcgjung** [comment]
@scherersebastian this is a PR to reinstall clean trivy and veracode scans in order to prepare the next release. Could you be so kind to take this over for review and merging?

My colleague @Martin0815bla will check/review for changes in the java logic (due to a changed config/property interface in EDC)

Thx.

**Review by Martin0815bla — APPROVED** [review]
Mostly version and white space changes

File: agent-plane/agentplane-azure-vault/src/main/docker/Dockerfile [hunk 1]
```diff
@@ -31,6 +31,8 @@
 ARG APP_UID=10100
 ARG APP_GID=30000
 
+RUN apk update && apk upgrade libssl3 libcrypto3 --no-cache
```
[the remark this answers is not in the document]
Comment (drcgjung) [reply agent-plane/agentplane-azure-vault/src/main/docker/Dockerfile:34]: no, its followed by RUN upgrade ... bummer

File: agent-plane/agentplane-hashicorp/src/main/docker/Dockerfile [hunk 2]
```diff
@@ -30,6 +30,8 @@
 ARG APP_UID=10100
 ARG APP_GID=30000
 
+RUN apk update && apk upgrade libssl3 libcrypto3 --no-cache
```
[the remark this answers is not in the document]
Comment (drcgjung) [reply agent-plane/agentplane-hashicorp/src/main/docker/Dockerfile:33]: see above. rule-based checking is only as good as the rule authors ...

**Comment by scherersebastian** [comment]
@drcgjung Overall, it looks very good :) 
I'm pretty underwater right now, so haven't checked the changes in context.

**Comment by drcgjung** [comment]
> @drcgjung Overall, it looks very good :) I'm pretty underwater right now, so haven't checked the changes in context.

@SebastianBezold could you take over. Sebastian Scherer and Martin checked already and I took back the apt upgrade/apt update thingy as you proposed in last week`s session.

**Review by SebastianBezold — APPROVED** [review]

**Pull request merged**